prov/util/src/util_attr.c

```diff
@@ -654,6 +654,23 @@ int ofi_check_domain_attr(const struct fi_provider *prov, uint32_t api_version,
 		return -FI_ENODATA;
 	}
 
+	if (user_attr->auth_key_size &&
+	    (user_attr->auth_key_size != prov_attr->auth_key_size)) {
```

Comment:
Based on @j-xiong's comment, lets change this check to
```
if (user_attr->auth_key && user_attr->auth_key_size &&
   (user_attr->auth_key_size != prov_attr->auth_key_size))
```
The below checks may not be needed then